cosmos/dbt/graph.py

```diff
@@ -223,10 +333,56 @@ def run_dbt_ls(
                 for line in logfile:
                     logger.debug(line.strip())
 
+        if self.should_use_dbt_ls_cache():
+            self.save_dbt_ls_cache(stdout)
+
         nodes = parse_dbt_ls_output(project_path, stdout)
         return nodes
 
     def load_via_dbt_ls(self) -> None:
+        """Retrieve the dbt ls cache if enabled and available or run dbt ls"""
+        if not self.load_via_dbt_ls_cache():
+            self.load_via_dbt_ls_without_cache()
+
+    @functools.lru_cache
+    def should_use_dbt_ls_cache(self) -> bool:
+        """Identify if Cosmos should use/store dbt ls cache or not."""
+        return settings.enable_cache and settings.enable_cache_dbt_ls and bool(self.dbt_ls_cache_key)
+
+    def load_via_dbt_ls_cache(self) -> bool:
+        """(Try to) load dbt ls cache from an Airflow Variable"""
+
+        logger.info(f"Trying to parse the dbt project using dbt ls cache {self.dbt_ls_cache_key}...")
+        if self.should_use_dbt_ls_cache():
+            project_path = self.project_path
+
+            cache_dict = self.get_dbt_ls_cache()
+            if not cache_dict:
+                logger.warning(f"Cosmos performance: Cache miss for {self.dbt_ls_cache_key}")
+                return False
+
+            cache_version = cache_dict.get("version")
+            dbt_ls_cache = cache_dict.get("dbt_ls")
+
+            current_version = cache.calculate_dbt_ls_cache_current_version(
+                self.dbt_ls_cache_key, project_path, self.dbt_ls_cache_key_args
+            )
+
+            if dbt_ls_cache and not cache.was_project_modified(cache_version, current_version):
+                logger.info(
+                    f"Cosmos performance [{platform.node()}|{os.getpid()}]: The cache size for {self.dbt_ls_cache_key} is {len(dbt_ls_cache)}"
+                )
+                self.load_method = LoadMode.DBT_LS_CACHE
+
+                nodes = parse_dbt_ls_output(project_path=project_path, ls_stdout=dbt_ls_cache)
+                self.nodes = nodes
+                self.filtered_nodes = nodes
+                logger.warning(f"Cosmos performance: Cache hit for {self.dbt_ls_cache_key} - {current_version}")
```

Comment:
```suggestion
                logger.info(f"Cosmos performance: Cache hit for {self.dbt_ls_cache_key} - {current_version}")
```
when cache is hit, do we still want it to be a warning?

Reply:
Nice catch, @pankajkoti ! I was just using that for troubleshooting, just reverted to info in: 97dddba00ac8cdb0e7a94a80d7c8230af66c990d